shared/chat/conversation/input-area/user-mention-hud/index.js

```diff
@@ -99,6 +100,10 @@ class MentionHud extends React.Component<MentionHudProps, State> {
 type ImplProps = MentionHudProps & {|data: Array<Data>, fullList: Array<Data>|}
 class MentionHudImpl extends React.Component<ImplProps> {
   componentDidUpdate(prevProps: ImplProps) {
+    if (this.props.filter !== prevProps.filter) {
```

Comment:
This works for the changing filter case, but when first typing '@' with the mouse nearby the immediate selection still happens.

Reply:
sigh, how this component works should change at a higher level. what comes in mutates every time anything changes so i have to do a deeper shallow compare. there's a todo in here to change how this works in general that we should do at some point soon

Reply:
ok i fixed this by initing the value (since component did update gets called after the first render). i'm not applying this to channel hud now. we need to clean up how this works first. i'll ticket

Reply:
Sounds good